crates/sui-graphql-rpc/src/types/balance.rs

```diff
@@ -14,3 +23,66 @@ pub(crate) struct Balance {
     /// Total balance across all coin objects of the coin type
     pub(crate) total_balance: Option<BigInt>,
 }
+
+type StoredBalance = (
+    /* balance */ Option<i64>,
+    /* count */ Option<i64>,
+    /* type */ Option<String>,
+);
+
+impl Balance {
+    /// Query for the balance of coins owned by `address`, of coins with type `coin_type`. Note that
+    /// `coin_type` is the type of `0x2::coin::Coin`'s type parameter, not the full type of the coin
+    /// object.
+    pub(crate) async fn query(
+        db: &Db,
+        address: SuiAddress,
+        coin_type: TypeTag,
+    ) -> Result<Option<Balance>, Error> {
+        use objects::dsl;
+
+        let stored: Option<StoredBalance> = db
+            .execute(move |conn| {
+                conn.first(move || {
+                    dsl::objects
+                        .select((
+                            sql::<Nullable<SqlBigInt>>("CAST(SUM(coin_balance) AS BIGINT)"),
```

Comment:
While working on benchmarking/stresstesting/raising limits, I found that this query will fail for the 0x0 address since we have a coin whose balance exceeds the max BIGINT value (it has 20 digits, while BIGINT caps out at 19). Could we bump the BIGINT type here up as you're touching the code?

Reply:
Oh interesting, good to know -- any idea what we should replace it with? If not, no worries, I can look into it.

Reply:
Fixed it in the Balance pagination PR, by reading it as `TEXT`!  I ended up not using `Numeric` or `Decimal` because they aren't supported in sqlite, but this suggests to me that we may need to fix this in other places places as well -- because we do generally treat it as OK to read a value as an `i64` and cast it to `u64` and this is an example where that's not okay.